### PR TITLE
Updating business rules component and feature

### PR DESCRIPTION
**Pull request by minudika**
This PR includes changes for following.
- Updating data source constants.
- Adding business-rule REST api.
- Bumping version of business-rules component and feature to 2.0.78-SNAPSHOT.

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/TemplateGroupsApi.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package org.wso2.carbon.business.rules.api;
+
+import org.wso2.carbon.business.rules.api.factories.TemplateGroupsApiServiceFactory;
+
+import io.swagger.annotations.ApiParam;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.*;
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/TemplateGroupsApi.java]: avoid using wild cards. use the exact package

File: components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/ApiOriginFilter.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
+package org.wso2.carbon.business.rules.api;
+
+import java.io.IOException;
+
+import javax.servlet.*;
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/ApiOriginFilter.java]: here too use exact packages

File: components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/ApiResponseMessage.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+@javax.xml.bind.annotation.XmlRootElement
+@javax.annotation.Generated(value = "io.swagger.codegen.languages.JavaMSF4JServerCodegen", date = "2017-09-20T06:15:20.154Z")
+public class ApiResponseMessage {
+    public static final int ERROR = 1;
+    public static final int WARNING = 2;
+    public static final int INFO = 3;
+    public static final int OK = 4;
+    public static final int TOO_BUSY = 5;
+
+    int code;
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/ApiResponseMessage.java]: we can use private scope right ?

File: components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/BusinessRuleApi.java [hunk 4]
```diff
@@ -0,0 +1,90 @@
+package org.wso2.carbon.business.rules.api;
+
+import org.wso2.carbon.business.rules.api.factories.BusinessRuleApiServiceFactory;
+import org.wso2.carbon.business.rules.model.BusinessRule;
+import org.wso2.carbon.business.rules.model.BusinessRuleDeploymentStatus;
+
+import io.swagger.annotations.ApiParam;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.*;
```
Comment (dnwick) [inline components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/BusinessRuleApi.java]: here too

Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/gen/java/org/wso2/carbon/business/rules/api/BusinessRuleApi.java]: These are auto generated classes by swagger. 
@ramindu90 : Should we change the scope?  ( how it happened in simulator)

**Pull request merged**